tests/clone_test.go

```diff
@@ -874,3 +729,148 @@ func createSourceVM(options ...libvmi.Option) (*virtv1.VirtualMachine, error) {
 	virtClient := kubevirt.Client()
 	return virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm, v1.CreateOptions{})
 }
+
+func createSnapshot(vm *virtv1.VirtualMachine) *snapshotv1.VirtualMachineSnapshot {
+	var err error
+
+	snapshot := &snapshotv1.VirtualMachineSnapshot{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "snapshot-" + vm.Name,
+			Namespace: vm.Namespace,
+		},
+		Spec: snapshotv1.VirtualMachineSnapshotSpec{
+			Source: k8sv1.TypedLocalObjectReference{
+				APIGroup: pointer.P(vmAPIGroup),
+				Kind:     "VirtualMachine",
+				Name:     vm.Name,
+			},
+		},
+	}
+
+	snapshot, err = kubevirt.Client().VirtualMachineSnapshot(snapshot.Namespace).Create(context.Background(), snapshot, v1.CreateOptions{})
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	return snapshot
+}
+
+func waitSnapshotReady(snapshot *snapshotv1.VirtualMachineSnapshot) *snapshotv1.VirtualMachineSnapshot {
```

Comment:
Can you return a new ptr instead of overwriting the one that was passed in?

Reply:
Possibly only pass in namespace and name of the snapshot?

Reply:
DONE